libs/auth/src/angular/registration/registration-finish/registration-finish.component.ts

```diff
@@ -94,6 +109,48 @@ export class RegistrationFinishComponent implements OnInit, OnDestroy {
     await this.router.navigate(["/login"], { queryParams: { email: this.email } });
   }
 
+  private async registerVerificationEmailClicked(email: string, emailVerificationToken: string) {
+    const request = new RegisterVerificationEmailClickedRequest(email, emailVerificationToken);
+
+    try {
+      const result = await this.accountApiService.registerVerificationEmailClicked(request);
+
+      if (result == null) {
+        this.toastService.showToast({
+          title: null,
+          message: this.i18nService.t("emailVerifiedV2"),
+          variant: "success",
+        });
+        this.loading = false;
+      }
+    } catch (e) {
+      await this.handleRegisterVerificationEmailClickedError(e);
+      this.loading = false;
+    }
+  }
+
+  private async handleRegisterVerificationEmailClickedError(e: unknown) {
+    if (e instanceof ErrorResponse) {
+      const errorResponse = e as ErrorResponse;
```

Comment:
🎨: I'm not sure the `switch` statement adds anything at this time. It looks like future proofing but maybe done too soon.
```suggestion
      const errorResponse = e as ErrorResponse;
      if (errorResponse.message.includes("Expired link")) {
        await this.router.navigate(["/signup-link-expired"]);
      } else {
        this.validationService.showError(errorResponse);
      }
```

Reply:
I like using a switch as it makes it easier to add additional error case handling in the future.

Reply:
¯\_(ツ)_/¯